legal-api/src/legal_api/core/meta/filing.py

```diff
@@ -332,8 +332,13 @@ def alter_outputs(filing_type: str, filing_meta_data: dict, outputs: set):
         elif filing_type == 'specialResolution' and 'changeOfName' in filing_meta_data.get('legalFilings', []):
             outputs.add('certificateOfNameChange')
         elif filing_type == 'correction':
+            filing_final = FILINGS.get(filing_type, {})
             if not filing_meta_data.get('correction', {}).get('toLegalName') and 'certificate' in outputs:
                 # For IA correction, certificate will be populated in get_all_outputs since
                 # legalFilings list contains correction and incorporationApplication
                 # and it should be removed if correction does not contain name change.
                 outputs.remove('certificate')
+            if filing_final.get('displayName') == 'Correction - Annual Report' and \
+                filing_final.get('codes') == 'CP' and 'noticeOfArticles' in outputs:
+                # For Coop filing, Correction - Annual Report and NOA, remove key: noticeOfArticles
+                outputs.remove('noticeOfArticles')
```

Comment:
This won't work. `displayName` in `filing_final` is always `Correction` and `codes` is not a string (`filing_final.get('codes') == 'CP'`). Checking display name is a bad idea. Look into this example https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/reports/report.py#L259 and get original_filing.filing_type

Reply:
Thanks. I've updated the code using the example. I also modified the method `alter_outputs` to better use the original filing instead of `filing_type` and `meta_data`.

Reply:
Cant we just remove CP from the line 
{'types': 'CP,BEN', 'outputs': ['noticeOfArticles', ]}, for correction?

Reply:
I dont think notice of articles is valid for CP ? Needs a confirmation from Linda

Reply:
@lekshmimallika-aot I agree. I don't find any NOA for CP in BE or FE either, but this ticket wants to suppress all NOA for correction, CP and annualReport.